### ISSUE TITLE
(quotemark) add option to enforce double quotes in JSX attributes and single quotes elsewhere

### ISSUE THREAD

**Issue by prabirshrestha** [palantir/tslint#673]

**Comment by adidahiya** [comment]
is this a feature request?

just to be clear, you want to enforce double quotes in HTML attributes within JSX?

```ts
<MyComponent someAttr="valid" another='invalid' />
```

**Comment by prabirshrestha** [comment]
This is a feature request.

```typescript
const value = 'some text';
<input placeholder="placeholder" value={value} />
```

I want to enforce double quotes only in attributes within JSX. If it isn't an attribute should use single quote.

**Comment by cajoy** [comment]
This rule is really needed for TSX files as tslinter complains about doublequotes in TSX files.

**Comment by nikolaymatrosov** [comment]
+1

**Comment by dupski** [comment]
+1

**Comment by JKillian** [comment]
Seems like a solid rule idea that is popular. I'll prioritize this!

**Comment by JKillian** [comment]
Just to double-check, this is just a stylistic issue right? Using single quotes for JSX attributes shouldn't cause any functional problems as far as I know:
```typescript
<div someProp='someValue' />
```

**Comment by dupski** [comment]
Hi Jason

Yes you can use either single or double quotes in JSX as far as I can tell,
so yes this is mainly a stylistic issue.

If possible I think it might be nice to configure JSX quoting separately
from TS, maybe a quotemark-jsx setting? There are instances where you might
want to interchangeably use the single or double quoting styles, e.g.
because it is not possible to escape quotes in JSX.

Cheers!

**Comment by JKillian** [comment]
I'm thinking we could add two more options to the quotemark rule: `jsx-single` and `jsx-double`. These rules would enforce either single or double quotes for JSX attributes. If you have the `avoid-escape` option set, they would allow you to use the other form of quotemark when necessary. Does this sound like it would meet everyone's use case?

**Comment by dupski** [comment]
Sounds good to me :)

**Comment by adidahiya** [comment]
this feature is available now in `tslint@next` (v3.2.2-dev.1)
